lighthouse-core/gather/driver.js

```diff
@@ -698,7 +697,6 @@ class Driver {
     const promise = new Promise((resolve, reject) => {
       checkForQuiet(this, resolve).catch(reject);
       cancel = () => {
-        cancelled = true;
```

Comment:
this feels like a bug, we want to just add `if (cancelled) return` right?

Reply:
oops, didn't meant to delete that.

can skip the guard here since reject/clearTimeout should have no effect after the first call.

Reply:
of course, leaving out the guard _looks_ like a mistake, so perhaps I should just add it anyways?

Reply:
> > can skip the guard here since reject/clearTimeout should have no effect after the first call.

> of course, leaving out the guard looks like a mistake, so perhaps I should just add it anyways?

I mean, it's also unnecessary everywhere else it's be added, isn't it? All removing listeners and clearing timeouts, which have no problem being run multiple times.

I'm OK with having them since that's how we would write a cancellable promise utility instead of doing each of these separately (in which case all of the methods should have it), but I'm also OK with dropping them since they aren't necessary (except for the `Security.disable` command in `_waitForSecurityCheck`)